lib/pbench/server/api/resources/users_api.py

```diff
@@ -199,17 +199,18 @@ def post(self):
         if not token_expiry:
             token_expiry = {"minutes": int(self.token_expire_duration)}
         elif type(token_expiry) is not dict:
-            self.logger.warning(
-                f"Invalid token expiry format {token_expiry}; type: {type(token_expiry)}"
-            )
             abort(
                 HTTPStatus.BAD_REQUEST,
-                message="Token expiry needs to be a dictionary",
+                message=(
+                    f"Invalid token expiry:  expected a JSON object, "
+                    f"got a {type(token_expiry)}"
+                ),
             )
-        elif any(key not in self.TOKEN_EXPIRY_KEYS for key in token_expiry.keys()):
+        bad_keys = sorted(set(token_expiry.keys()) - self.TOKEN_EXPIRY_KEYS)
+        if bad_keys:
             msg = (
-                f"Invalid token expiry key: found {set(token_expiry.keys())}, "
-                f"expected of one {self.TOKEN_EXPIRY_KEYS}"
+                f"Invalid token expiry key{'s' if len(bad_keys) > 1 else ''}: "
+                f"found {bad_keys}; expected one of {sorted(self.TOKEN_EXPIRY_KEYS)}"
             )
             self.logger.warning(msg)
             abort(
```

Comment:
This is a good try, but I don't think you'll like it:  if the block at line 200 is executed, we'll still construct and evaluate `bad_keys` -- that won't trigger an abort, but it's wasted time.  If you really want to use the common subexpression, you need to do something like the following:
```
        if not token_expiry:
            token_expiry = {"minutes": int(self.token_expire_duration)}
        elif type(token_expiry) is not dict:
            abort(
                HTTPStatus.BAD_REQUEST,
                message=(
                    f"Invalid token expiry:  expected a JSON object, "
                    f"got a {type(token_expiry)}"
                ),
            )
        else:
            bad_keys = sorted(set(token_expiry.keys()) - self.TOKEN_EXPIRY_KEYS)
            if bad_keys:
                msg = (
                    f"Invalid token expiry key{'s' if len(bad_keys) > 1 else ''}: "
                    f"found {bad_keys}; expected one of {sorted(self.TOKEN_EXPIRY_KEYS)}"
                )
                self.logger.warning(msg)
                abort(
                    ...
```
and that introduces another level of indentation, which will probably make Black do terrible things to `msg`, etc.  (Which is why I figured that repeating the set-difference operation so that I could use an `elif` was the lesser of the two weevils.)